src/Pmi/Datastore/DatastoreClientHelper.php

```diff
@@ -22,9 +22,12 @@ public function __construct()
         ]);
     }
 
-    public function fetchAll($kind)
+    public function fetchAll($kind, $limit = null)
     {
         $query = $this->datastore->query()->kind($kind);
+        if ($limit) {
+            $query = $query->limit($limit);
```

Comment:
The Query methods mutate the object, so you don't need to assign it back to $query. The methods return itself for chaining purposes. In this case, you just need to call `$query->limit($limit);`